CHANGELOG.md

```diff
@@ -43,6 +43,16 @@ and this project adheres to
   T2S template to set bit 27 of `MSR_IA32_ARCH_CAPABILITIES` (`RFDS_NO`) to 1
   since it assumes that the fleet only consists of processors that are not
   affected by RFDS.
+- [#4388](https://github.com/firecracker-microvm/firecracker/pull/4388): Avoid
+  setting `kvm_immediate_exit` to 1 if are already handling an exit, or if the
+  vCPU is stopped. This avoids a spurious KVM exit upon restoring snapshots.
+- [#4567](https://github.com/firecracker-microvm/firecracker/pull/4567): Do not
+  initialize vCPUs in powered-off state upon snapshot restore. No functional
+  change, as vCPU initialization is only relevant for the booted case (where the
+  guest expects CPUs to be powered off).
+- [#4584](https://github.com/firecracker-microvm/firecracker/pull/4584): On ARM,
+  always store the `kvm_vcpu_init` structure in snapshots instead of only when
+  CPU templates were used. This makes snapshot handling for these cases uniform.
```

Comment:
I still can't see an externally visible side effect, to be honest. This is only about code structure, isn't it?

Reply:
Mh, I guess the snapshots in the non-cpu-template case will be ever so slightly larger. Maybe I change "instead of only when CPU template were used" add a "instead of recomputing it on snapshot restore if no CPU templates were used"?